pkg/controller/controller_test.go

```diff
@@ -1857,42 +2111,53 @@ func getRecordedEvents(testController *controller) []string {
 	return events
 }
 
-func assertNumEvents(t *testing.T, strings []string, number int) {
+func assertNumEvents(strings []string, number int) error {
 	if e, a := number, len(strings); e != a {
-		t.Fatalf("Unexpected number of events: expected %v, got %v", e, a)
+		return fmt.Errorf("Unexpected number of events: expected %v, got %v", e, a)
 	}
+
+	return nil
 }
 
-func assertNumberOfActions(t *testing.T, actions []clientgotesting.Action, number int) {
+func assertNumberOfActions(t *testing.T, actions []clientgotesting.Action, number int) error {
```

Comment:
I'm confused, do we not get a stack trace showing the originating test?

If not, it seems like we don't really achieve anything here besides the one log.